src/sinks/util/encoding/encodings.rs

```diff
@@ -0,0 +1,74 @@
+/// Enums representing different ways to encode events as they are sent into a Sink.
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Copy, Debug, Derivative, Deserialize, Eq, PartialEq, Serialize)]
+#[derivative(Default)]
+#[serde(rename_all = "snake_case")]
+pub enum EncodingText {
```

Comment:
Hmmm I guess the alternative here would be to have a single `Encoding` type with no default, then make each component implement a serde default function? It might adapt better as we add things like https://github.com/timberio/vector/pull/5021

Reply:
:+1: I think I like this better too.